src/main/java/org/opentest4j/fixtures/ClassWrapper.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j.fixtures;
+
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+import static org.opentest4j.fixtures.Utils.toUnmodifiableList;
+
+import java.lang.annotation.Annotation;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Simple wrapper around {@link Class}.
+ *
+ * @author Kevin Cooney
+ * @since 1.1
+ */
+public final class ClassWrapper {
+    private final ReentrantLock lock = new ReentrantLock();
+    private final Class<?> wrappedClass;
+    private volatile List<Annotation> annotations;
+
+	public ClassWrapper(Class<?> classToWrap) {
+		wrappedClass = requireNonNull(classToWrap);
+	}
+
+	public ClassWrapper(Class<?> classToWrap, List<Annotation> classAnnotations) {
+		wrappedClass = requireNonNull(classToWrap, "classToWrap cannot be null");
+		annotations = toUnmodifiableList(requireNonNull(classAnnotations, "classAnnotations cannot be null"));
+	}
```

Comment:
@kcooney If this PR is approved, you should strongly consider making these constructors _static factory methods_, à la Effective Java 2nd Edition, Item 1.

For example:
- `ClassWrapper.of(Class<?>)`
- `ClassWrapper.withAnnotations(Class<?>, List<Annotation>)`.